### DISCUSSION TITLE
Definition of productive clones

### DISCUSSION THREAD

**Discussion by rvalieris** [milaboratory/mixcr#1549] [Q&A] [answered]
Hello,

I'm trying to replicate the `productive` column from the airr format with the exportClones command.

exportClones have a `-isProductive <geneFeature>` flag, which gene feature should I use here to have the same output as the exportAirr command ?

from what I tested it seems to be CDR3, but I wanted to confirm this, and then also ask what the flag `--export-productive-clones-only` is doing because that gives me different results.

**Comment by mizraelson** [comment]
Hi,
Yes, the correct feature is `CDR3`. Can you give an example where using `--export-productive-clones-only` gives a different result?

**Comment by rvalieris** [comment]
oh I see, I might be doing something wrong then, here is what I tried:

```bash
PREFIX=SRR4026007 # https://ncbi.nlm.nih.gov/sra/?term=SRR4026007
R1="$PREFIX"_1.fastq.gz
R2="$PREFIX"_2.fastq.gz

mixcr -Xmx10G analyze generic-amplicon-with-umi \
  --limit-input 10000 \
  --no-json-reports \
  --species hs \
  --rna \
  --tag-pattern '^(R1:*) \ ^(UMI:N{17})(R2:*)' \
  --floating-left-alignment-boundary \
  --floating-right-alignment-boundary C \
  --assemble-clonotypes-by '{CDR1Begin:FR4End}' \
  $R1 $R2 $PREFIX

mixcr exportClones --chains IGH -isProductive CDR3 -hasStops CDR3 -isOOF CDR3 $PREFIX.clns test1.tsv
mixcr exportClones --export-productive-clones-only --chains IGH -isProductive CDR3 -hasStops CDR3 -isOOF CDR3 $PREFIX.clns test2.tsv
```

then I count how many clones are in each file, according to the isProductive column:
```bash
$ cut -f35 test1.tsv | sort | uniq -c
     22 false
      1 isProductiveCDR3
    100 true

$ cut -f35 test2.tsv | sort | uniq -c
      1 isProductiveCDR3
     91 true

$ cut -f36 test1.tsv | sort | uniq -c
    122 false
      1 hasStopsInCDR3

$ cut -f36 test2.tsv | sort | uniq -c
     91 false
      1 hasStopsInCDR3

$ cut -f37 test1.tsv | sort | uniq -c
    100 false
      1 isOOFCDR3
     21 region_not_covered
      1 true

$ cut -f37 test2.tsv | sort | uniq -c
     91 false
      1 isOOFCDR3
```

I expected test2 would have then same 100 productive clones as in test1.

**Reply by mizraelson** [reply, marked as answer]
Hi,

Yes, the --export-productive-clones-only option examines the entire region, not solely the 'CDR3'. Typically, we anticipate that stop codons or frame shifts would occur exclusively within the CDR3 region and the outputs are the sane. However, in the dataset you shared (SRR4026007), some clones have hypermutations resulting in stop codons appearing in regions outside of CDR3. Hence the differences.

**Reply by rvalieris** [reply]
> the --export-productive-clones-only option examines the entire region, not solely the 'CDR3'

ah ok, that makes sense, thanks.
